build_projects/dotnet-cli-build/PublishTargets.cs

```diff
@@ -19,160 +19,115 @@ public static class PublishTargets
-                 { "Windows_x86", false },
-                 { "Windows_x64", false },
-                 { "Ubuntu_x64", false },
-                 { "RHEL_x64", false },
-                 { "OSX_x64", false },
-                 { "Debian_x64", false },
-                 { "CentOS_x64", false }
+                 { "build-Windows_x86", false },
+                 { "build-Windows_x64", false },
+                 { "build-Ubuntu_x64", false },
+                 { "build-RHEL_x64", false },
+                 { "build-OSX_x64", false },
+                 { "build-Debian_x64", false },
+                 { "build-CentOS_x64", false }
              };
 
-            List<string> blobs = new List<string>(AzurePublisherTool.ListBlobs($"{Channel}/Binaries/{CliNuGetVersion}/"));
-
-            var versionBadgeName = $"{CurrentPlatform.Current}_{CurrentArchitecture.Current}";
-            if (badges.ContainsKey(versionBadgeName) == false)
-            {
-                throw new ArgumentException("A new OS build was added without adding the moniker to the {nameof(badges)} lookup");
-            }
-
-            foreach (string file in blobs)
-            {
-                string name = Path.GetFileName(file);
-                string key = string.Empty;
-
-                foreach (string img in badges.Keys)
-                {
-                    if ((name.StartsWith($"{img}")) && (name.EndsWith(".svg")))
-                    {
-                        key = img;
-                        break;
-                    }
-                }
-
-                if (string.IsNullOrEmpty(key) == false)
-                {
-                    badges[key] = true;
-                }
-            }
+             IEnumerable<string> blobs = AzurePublisherTool.ListBlobs(AzurePublisher.Products.SDK, CliVersion);
+             foreach (string blob in blobs)
+             {
+                 if (blob.StartsWith("build"))
+                 {
+                     string name = blob.Substring(0, blob.IndexOf("."));
+                     if (monikers.ContainsKey(name))
+                     {
+                         monikers[name] = true;
+                     }
+                     else
+                     {
+                         throw new ArgumentOutOfRangeException("Unknown OS moniker found during build finalization");
+                     }
+                 }
+             }
+
+             return monikers.Keys.All(k => monikers[k]);
+        }
+        
+        public static void PublishCliVersionInformation(BuildTargetContext c)
+        {
+            var versionBadge = c.BuildContext.Get<string>("VersionBadge");
+            VersionUpdater.PublishVersionBadge(versionBadge, Channel, Pointer).Wait();
+            VersionUpdater.PublishVersionFile("build.version", Utils.GetCliVersionFileContent(c), Channel, Pointer).Wait();
+        }
 
-            return badges.Keys.All(key => badges[key]);
+        public static void PublishCliNupkgs(BuildTargetContext c)
```

Comment:
Currently the Windows x64 VSO build publishes the NuGet packages after the "Publish" step.

We either need to:

1. continue updating dotnet/versions of the new packages AFTER the packages have actually been published
2. move the publishing of the nupkgs into the FinalizeBuild step, and ensure that we ONLY publish the Windows x64 nupkgs.
 - I imagine this working like:  
 - Windows x64 knows that it should publish the .nupkgs to Azure
 - FinalizeBuild downloads those packages and then publishes them to MyGet

We shouldn't be updated dotnet/versions before the packages have actually been published to MyGet (in case they never get published).

---
In reply to: [64946020](https://github.com/dotnet/cli/pull/3270#discussion_r64946020) [](ancestors = 64946020)